### PR TITLE
[sw/test_rom] Use SPI Flash based bootstrap in test_rom

### PR DESCRIPTION
**Pull request by alphan**
This PR removes the device-side code for the previous bootstrap implementation and replaces its use in test_rom with the SPI Flash based bootstrap.

Since english breakfast is missing otp, life cycle controller, and reset manager, I ended up adding some fake functions for them. This slightly complicates the build files but I prefer this over polluting silicon_creator or test_rom source code.

Created #12905.

### CODE REVIEW

**Comment by alphan** [comment]
@weicaiyang @tjaychen @msfschaffner This change will cause any sims that bootstrap the chip over SPI to fail as we discussed in the chip-level testing meeting (I think there was only one).

**Comment by vogelpi** [comment]
Hi @alphan ,
we still need the CW305/English Breakfast, see #12905 for details.

**Comment by alphan** [comment]
> Hi @alphan , we still need the CW305/English Breakfast, see #12905 for details.

Thanks, I'll take another look at this when I get the chance.

**Comment by alphan** [comment]
Tested this change locally on the english breakfast top level on a CW305 using a simple python script that I've put together. I was able to bootstrap with `aes_smoketest_fpga_nexysvideo.bin` and got the following on UART:
```
I00000 test_rom.c:81] Version:    opentitan-earlgrey_silver_release_v5-5700-g8fc376f3a
Build Date: 2022-06-02, 10:54:11

I00001 test_rom.c:87] TestROM:132cb64f
I00002 test_rom.c:118] Test ROM complete, jumping to flash!
I00000 aes_smoketest.c:62] Running AES test
E00001 ottf_isrs.c:54] FAULT: Store Access Fault. MCAUSE=00000007 MEPC=20002952 MTVAL=41160020
```

**Comment by alphan** [comment]
@vogelpi I updated the PR, should be ready to review. We can now use the same bootstrap on english breakfast as well. PTAL.

P.S. I'll create a separate PR in lowRISC/ot-sca for a new bootstrap script.

File: sw/device/lib/testing/test_rom/english_breakfast_fake_driver_funcs.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+
+// TODO(#12905): This file includes fake definitions for the functions that are
+// used by the silicon_creator bootstrap implementation but missing in the
+// english breakfast top level due to hardware limitations.
+
+#if !OT_IS_ENGLISH_BREAKFAST
+static_assert(
```
Comment (moidx) [inline sw/device/lib/testing/test_rom/english_breakfast_fake_driver_funcs.c:16]: nit: Also `#error "error msg"` but I assume `static_assert` is more consistent with the codebase.

**Review by moidx — APPROVED** [review]

**Review by cfrantz — APPROVED** [review]

File: sw/device/lib/testing/test_rom/english_breakfast_fake_driver_funcs.c (hunk 1 above)
Comment (alphan) [reply sw/device/lib/testing/test_rom/english_breakfast_fake_driver_funcs.c]: Good point, done, thanks!

**Comment by alphan** [comment]
@drewmacrae @tjaychen The last commit should fix the issue that I mentioned earlier. The bazel target was using the wrong autogen header which defines the write window size as 512 bytes as opposed to 64 bytes. That was the reason for the error from `flash_ctrl_data_write()`.

I'm not sure what's the best way to avoid something similar in the future and make sure that all dependencies are correct but maybe we should start with a comparison with `/meson.build`

**Comment by alphan** [comment]
> LGTM, thanks @alphan. If you could help out with the bootstrapping script in ot-sca this would be greatly appreciated.

My pleasure, please see: lowRISC/ot-sca#75

**Comment by a-will** [comment]
It'd be nice if we kept the Python script up to date as well. The opentitantool build takes an inconveniently long time, and testing changes across different commits can easily trigger rebuilds.

**Comment by vogelpi** [comment]
I agree that we should keep the Python script working. It's still how people are supposed to operate the FPGA (see documentation!).

**Comment by alphan** [comment]
> I agree that we should keep the Python script working. It's still how people are supposed to operate the FPGA (see [documentation](https://docs.opentitan.org/doc/getting_started/setup_fpga/)!

Thanks bringing up the documentation issue @vogelpi! #13019 updates the references to the loader script and I believe we'll have additional docs describing bazel and opentitantool.

**Pull request merged**